code/espurna/api.h

```diff
@@ -22,14 +22,71 @@ String apiKey();
 
 #if WEB_SUPPORT && API_SUPPORT
 
-#include <functional>
+#include <vector>
 
-using api_get_callback_f = std::function<void(char * buffer, size_t size)>;
-using api_put_callback_f = std::function<void(const char * payload)> ;
+constexpr unsigned char ApiUnusedArg = 0u;
 
-void apiRegister(const String& key, api_get_callback_f getFn, api_put_callback_f putFn = nullptr);
+struct ApiBuffer {
+    constexpr static size_t size = API_BUFFER_SIZE;
+    char data[size];
+
+    void erase() {
+        std::fill(data, data + size, '\0');
+    }
+};
+
+struct Api {
+    using BasicHandler = void(*)(const Api& api, ApiBuffer& buffer);
+    using JsonHandler = void(*)(const Api& api, JsonObject& root);
+
+    enum class Type {
+        Basic,
+        Json
+    };
```

Comment:
Initial attempt also included Type::Indexed
This is just 'nice-to-have' atm, since it only benefits relays and light channels, as sensor magnitudes are all named differently most of the time. However, this should be an easy change:
- adding ApiMatch check before comparing paths, when Api object includes `#` (just like `mqttMagnitude()`)
- setting `Api::arg` to the value extracted from the path, thus using ApiUnusedArg placeholder
- (or) 3rd function type that includes index arg instead of re-using `Api::arg`